pkg/kubelet/lifecycle/predicate.go

```diff
@@ -64,6 +74,18 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		}
 	}
 	if !fit {
+		fit, reasons, err = w.admissionFailureHandler.HandleAdmissionFailure(pod, reasons)
+		if err != nil {
+			message := fmt.Sprintf("HandleAdmissionFailure failed due to %v, which is unexpected.", err)
```

Comment:
including some info about expected outcome/output would be very helpful in debugging